cli/src/lib/proxy.js

```diff
@@ -0,0 +1,127 @@
+const url = require('url')
+const { reporter } = require('@dhis2/cli-helpers-engine')
+const httpProxy = require('http-proxy')
+const _ = require('lodash')
+const transformProxyResponse = require('node-http-proxy-json')
+
+const stripCookieSecure = cookie => {
+    return cookie
+        .split(';')
+        .filter(v => v.trim().toLowerCase() !== 'secure')
+        .join('; ')
+}
+
+const rewriteLocation = ({ location, target, baseUrl }) => {
+    const parsedLocation = url.parse(location)
+    const parsedTarget = url.parse(target)
+    const parsedBaseUrl = url.parse(baseUrl)
+
+    if (
+        parsedLocation.host === parsedTarget.host &&
+        parsedLocation.pathname.startsWith(parsedTarget.pathname)
+    ) {
+        return url.format({
+            ...parsedBaseUrl,
+            pathname: parsedLocation.pathname.replace(
+                parsedTarget.pathname,
+                ''
+            ),
```

Comment:
Won't this have issues if i.e. the target pathname is `/dev` and there's a `/dev` somewhere OTHER than at the start of the string?  I.e. we should probably replace `/^${pathname}/` instead, so that only the first match is replaced

Reply:
When a string is passed to `String.prototype.replace`, only the first match is replaced (e.g. `'aaa'.replace('a', 'b') === 'baa'`). For it to replace multiple matches, a regex with the `g` flag must be passed (`'aaa'.replace(/a/g, 'b') === 'bbb'`).